File: Server/src/main/java/com/api/server/foodb/Food.java
package com.api.server.foodb;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;

import java.time.Instant;

@Getter
@Setter
@Entity
@Table(name = "foods")
public class Food {
    @Id
    @Column(name = "id", nullable = false)
    private Integer id;

    @Size(max = 255)
    @NotNull
    @Column(name = "name", nullable = false)
    private String name;

    @Size(max = 255)
    @Column(name = "name_scientific")
    private String nameScientific;

    @Lob
    @Column(name = "description")
    private String description;

    @Size(max = 255)
    @Column(name = "itis_id")
    private String itisId;

    @Size(max = 255)
    @Column(name = "wikipedia_id")
    private String wikipediaId;

    @Size(max = 255)
    @Column(name = "picture_file_name")
    private String pictureFileName;

    @Size(max = 255)
    @Column(name = "picture_content_type")
    private String pictureContentType;

    @Column(name = "picture_file_size")
    private Integer pictureFileSize;

    @Column(name = "picture_updated_at")
    private Instant pictureUpdatedAt;

    @Column(name = "legacy_id")
    private Integer legacyId;

    @Size(max = 255)
    @Column(name = "food_group")
    private String foodGroup;

    @Size(max = 255)
    @Column(name = "food_subgroup")
    private String foodSubgroup;

    @Size(max = 255)
    @NotNull
    @Column(name = "food_type", nullable = false)
    private String foodType;

    @Column(name = "created_at")
    private Instant createdAt;

    @Column(name = "updated_at")
    private Instant updatedAt;

    @Column(name = "creator_id")
    private Integer creatorId;

    @Column(name = "updater_id")
    private Integer updaterId;

    @NotNull
    @Column(name = "export_to_afcdb", nullable = false)
    private Boolean exportToAfcdb = false;

    @Size(max = 255)
    @Column(name = "category")
    private String category;

    @Column(name = "ncbi_taxonomy_id")
    private Integer ncbiTaxonomyId;

    @Column(name = "export_to_foodb")
    private Boolean exportToFoodb;

    @Size(max = 255)
    @Column(name = "public_id")
    private String publicId;

}

File: DotNetServer/DotNetServer/FoodModule/Entities/Food.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace DotNetServer.FoodModule.Entities;

[Table("foods")]
[Index("ExportToAfcdb", Name = "index_foods_on_export_to_afcdb")]
[Index("Name", Name = "index_foods_on_name", IsUnique = true)]
[Index("NameScientific", Name = "index_foods_on_name_scientific")]
public partial class Food
{
    [Key] [Column("id")] public int Id { get; set; }

    public List<Content> Contents { get; set; } = null!;

    [Column("name")] public string Name { get; set; } = null!;

    [Column("name_scientific")] public string? NameScientific { get; set; }

    [Column("description", TypeName = "mediumtext")]
    public string? Description { get; set; }

    [Column("itis_id")]
    [StringLength(255)]
    public string? ItisId { get; set; }

    [Column("wikipedia_id")]
    [StringLength(255)]
    public string? WikipediaId { get; set; }

    [Column("picture_file_name")]
    [StringLength(255)]
    public string? PictureFileName { get; set; }

    [Column("picture_content_type")]
    [StringLength(255)]
    public string? PictureContentType { get; set; }

    [Column("picture_file_size")] public int? PictureFileSize { get; set; }

    [Column("picture_updated_at", TypeName = "datetime")]
    public DateTime? PictureUpdatedAt { get; set; }

    [Column("legacy_id")] public int? LegacyId { get; set; }

    [Column("food_group")]
    [StringLength(255)]
    public string? FoodGroup { get; set; }

    [Column("food_subgroup")]
    [StringLength(255)]
    public string? FoodSubgroup { get; set; }

    [Column("food_type")]
    [StringLength(255)]
    public string FoodType { get; set; } = null!;

    [Column("created_at", TypeName = "datetime")]
    public DateTime? CreatedAt { get; set; }

    [Column("updated_at", TypeName = "datetime")]
    public DateTime? UpdatedAt { get; set; }

    [Column("creator_id")] public int? CreatorId { get; set; }

    [Column("updater_id")] public int? UpdaterId { get; set; }

    [Column("export_to_afcdb")] public bool ExportToAfcdb { get; set; }

    [Column("category")]
    [StringLength(255)]
    public string? Category { get; set; }

    [Column("ncbi_taxonomy_id")] public int? NcbiTaxonomyId { get; set; }

    [Column("export_to_foodb")] public bool? ExportToFoodb { get; set; }

    [Column("public_id")]
    [StringLength(255)]
    public string? PublicId { get; set; }
}